src/main/game/Game.java

```diff
@@ -1,39 +1,95 @@
 package main.game;
 
+import java.util.HashMap;
+
 public class Game {
+    private final English english;
+    private final Greek greekLanguage;
     private Rules rules;
     private UI inputOutput;
 
-    public Game(UI inputOutput, Rules rules) {
+    public Game(UI inputOutput, Rules rules, English englishLanguage, Greek greekLanguage) {
         this.inputOutput = inputOutput;
         this.rules = rules;
+        this.english = englishLanguage;
+        this.greekLanguage = greekLanguage;
     }
 
-    public void runGame() {
-        inputOutput.askUserForGameMode();
-        String gameMode = inputOutput.getGameMode();
+    public String getGameMode(Language language) {
+        inputOutput.askUserForGameMode(language);
+        return inputOutput.getGameMode();
+    }
 
+    public String getLanguageSelection() {
+        inputOutput.askUserForLanguageSelection();
+        return inputOutput.getLanguageSelection();
+    }
+
+    public void runGame(Language language, String gameMode) {
+        Moves playerOneMove = getPlayerMove(language, language.playerOneType());
+        Moves playerTwoMove = getPlayerTwoMove(language, gameMode);
+        findWinner(playerOneMove, playerTwoMove, language);
+    }
+
+    public Moves getPlayerTwoMove(Language language, String gameMode) {
         if (gameMode.equals("1")) {
-            Moves playerOneMove = getPlayerMove("Player one");
-            Moves playerTwoMove = getPlayerMove("Player two");
-            findWinner(playerOneMove, playerTwoMove);
+            return getPlayerMove(language, gameMode);
         } else {
-            Moves playerOneMove = getPlayerMove("Player one");
-            Moves playerTwoMove = ComputerPlayer.playRandomMove();
-            inputOutput.displayComputerMove(playerTwoMove.getMove());
-            findWinner(playerOneMove, playerTwoMove);
+            Moves computerMove = ComputerPlayer.playRandomMove();
+            if (language.equals(greekLanguage)) {
```

Comment:
This is an explicit type-check in disguise. Can you make this polymorphic behaviour instead of doing this type of check?